### ISSUE TITLE
Expose raw data and support queries with geometry

### ISSUE THREAD

**Issue by dionhaefner** [DHI-GRAS/terracotta#244]
I'm not against adding a route mapping `[bbox, zoom level] -> png` and `[bbox, zoom level] -> pixel data`. Something like

```
# vanilla singleband tile
/singleband/{keys}/{x}/{y}/{z}.png

# untiled access with bbox
/singleband/{keys}/{z}.png?bbox={wkt}

# equivalent to current /preview.png if z == native zoom level (bbox = extent)
/singleband/{keys}/{z}.png

# raw pixel data of tile instead of png image
/singleband/{keys}/{x}/{y}/{z}.json

# raw pixel data in bbox
/singleband/{keys}/{z}.json?bbox={bbox}

# raw pixel data via compute, as requested in this issue
/compute/{keys}/{z}.json?expression={expression}&bbox={bbox}

# also supports reductions like zonal mean
/compute/{keys}/{z}.json?expression=mean(v1)&bbox={bbox}&v1=foo
```

With lots of validation on the passed bbox. In this proposal all raw pixel data would be *aligned with the XYZ grid*, so there would be no way to read the unprojected raw data as in the COG. But this also allows for efficient usage of overviews and makes it so we don't have to inform the user of which CRS the data is in, and ensures that tiles and raw data match.

So the `.json` routes just give you whatever the `.png` would give without the final step of rendering the data as an image (basically the raw 2D or 3D NumPy array as text).

_Originally posted by @dionhaefner in https://github.com/DHI-GRAS/terracotta/issues/226#issuecomment-881291930_

**Comment by j08lue** [comment]
We have had this on an internal wish list for a while, since pre-fetched PNG for a selected area is a way to render a smooth animation of a selected area on the map.

**Comment by j08lue** [comment]
Yes, and binary tiles are a cool concept, too. Btw, we have this simple client library https://github.com/DHI-GRAS/terracotta-react
